nixops/deployment.py

```diff
@@ -137,11 +158,39 @@ def get_machine(self, name):
             raise Exception("resource ‘{0}’ is not a machine".format(name))
         return res
 
+    def _definitions(self) -> Definitions:
+        if self.definitions is None:
+            raise Exception("Bug: Deployment.definitions is None.")
+        return self.definitions
+
+    def _definition_for(
+        self, name: str
+    ) -> Optional[nixops.resources.ResourceDefinition]:
+        definitions = self._definitions()
+
+        return definitions[name]
+
+    def _definition_for_required(
+        self, name: str
+    ) -> nixops.resources.ResourceDefinition:
+        defn = self._definition_for(name)
+        if defn is None:
+            raise Exception("Bug: Deployment.definitions['{}'] is None.".format(name))
+        return defn
+
+    def _machine_definition_for_required(
+        self, name: str
+    ) -> nixops.backends.MachineDefinition:
+        defn = self._definition_for_required(name)
+        if not isinstance(defn, nixops.backends.MachineDefinition):
+            raise Exception("Definition named '{}' is not a machine.".format(name))
+        return defn
```

Comment:
Under the old behavior:

1. self.definitions was None, causing `self.definititions[name]` to be an Exception -- no change
2. self.definitions was a Dict, causing `self.definititions[name]` to be possibly None -- generally not causing an Exception, but instead propogating a None around
3. self.definitions was a Dict, `self.definititions[name]` was something, but not a Machine -- instead a different kind of resource. I believe this would cause an exception.

In other words, the only case I'm somewhat unsure about is the None case.